parquet/src/test/java/org/apache/iceberg/parquet/TestDictionaryRowGroupFilter.java

```diff
@@ -1030,7 +1030,7 @@ public void testMissingDictionaryPageForColumn() {
         IllegalStateException.class,
         "Failed to read required dictionary page for id: 5",
         () ->
-            new ParquetDictionaryRowGroupFilter(SCHEMA, notEqual("some_nulls", "some"))
+            new ParquetDictionaryRowGroupFilter(SCHEMA, equal("some_nulls", "some"))
```

Comment:
The old UT will fail after this change because `notEq` will not try to read the dictionary when there may exist null values. Fixed it by using `eq` as the predicate.

Reply:
I'm not sure it's expected behavior then.  Looks like original expectation to fail if there is no dictionary value, regardless of whether there are nulls or not?  I'm not familiar, maybe worth to check with original author @rdblue

Reply:
Thanks for reviewing!

I checked the PR #93 that introduced this UT and the relevant [discussion](https://github.com/apache/iceberg/pull/86#discussion_r253802993). Basically it wants to throw an exception when no dict is available to avoid NPE after calling `dict`. But as @rdblue mentioned [here](https://github.com/apache/iceberg/pull/86#discussion_r253943635),  all runs of `dict` in `ParquetDictionaryRowGroupFilter` is guaranteed that there must be a dictionary. So under normal circumstances, this exception will not be thrown out I think. The reason why this UT can catch the exception is because it mocked an abnormal [DictionaryPageReadStore](https://github.com/apache/iceberg/blob/49d833aa83f6d4681c6c9f8473080f6e78124a0e/parquet/src/test/java/org/apache/iceberg/parquet/TestDictionaryRowGroupFilter.java#L1034).
Above is the context about this UT. So I think replacing it with `eq` can still serve the original purpose of verification, right? So I think moving null-check is reasonable because the run of `notEq` or `notIn` is guaranteed there will be a dictionary there. Moving it forward can help avoiding the dictionary decoding overhead when the column has nulls.